test/parallel/test-child-process-exec-kill-throws.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+// Flags: --expose_internals
+const common = require('../common');
+const assert = require('assert');
+const cp = require('child_process');
+const internalCp = require('internal/child_process');
+
+if (process.argv[2] === 'child') {
+  // Keep the process alive and printing to stdout.
+  setInterval(() => { console.log('foo'); }, 1);
+} else {
+  // Monkey path ChildProcess#kill() to kill the process and then throw.
+  const kill = internalCp.ChildProcess.prototype.kill;
+
+  internalCp.ChildProcess.prototype.kill = function() {
+    kill.apply(this, arguments);
+    throw new Error('mock error');
+  };
+
+  const cmd = `${process.execPath} ${__filename} child`;
+  const options = { maxBuffer: 0 };
+  const child = cp.exec(cmd, options, common.mustCall((err, stdout, stderr) => {
+    // Verify that if ChildProcess#kill() throws, the error is reported.
```

Comment:
I'm not sure that I understand the question. Do you mean if Node decides to kill the child process, but in the interim it exits on its own? If `exithandler()` runs (via the `close` handler for example), then the error won't be reported because it contains a check to make sure it doesn't run twice.

FWIW, [this](https://github.com/nodejs/node/blob/3268863ebc40d1f0beee61b044c492b43fa57fa5/lib/child_process.js#L262-L267) is the code path being tested.